src/topp/FileMerger.cpp

```diff
@@ -309,65 +359,60 @@ class TOPPFileMerger :
           }
           else if (rt_filename)
           {
-            if (!filename.hasSubstring("rt"))
+            static const boost::regex re("rt(\\d+(\\.\\d+)?)");
+            boost::smatch match;
+            bool found = boost::regex_search(filename, match, re);
+            if (found)
             {
-              writeLog_(String("Warning: cannot guess retention time from filename as it does not contain 'rt'"));
+              rt_final = String(match[1]).toFloat();
             }
-            for (Size i = 0; i < filename.size(); ++i)
+            else
             {
-              if (filename[i] == 'r' && ++i != filename.size() && filename[i] == 't' && ++i != filename.size() && isdigit(filename[i]))
-              {
-                String rt;
-                while (i != filename.size() && (filename[i] == '.' || isdigit(filename[i])))
-                {
-                  rt += filename[i++];
-                }
-                if (rt.size() > 0)
-                {
-                  // remove dot from rt3892.98.dta
-                  //                          ^
-                  if (rt[rt.size() - 1] == '.')
-                  {
-                    // remove last character
-                    rt.erase(rt.end() - 1);
-                  }
-                }
-                try
-                {
-                  float tmp = rt.toFloat();
-                  rt_final = tmp;
-                }
-                catch (Exception::ConversionError)
-                {
-                  writeLog_(String("Warning: cannot convert the found retention time in a value '" + rt + "'."));
-                }
-              }
+              writeLog_("Warning: could not extract retention time from filename '" + filename + "'");
             }
           }
 
           // none of the rt methods were successful
-          if (rt_final == -1)
+          if (rt_final < 0)
           {
             writeLog_(String("Warning: No valid retention time for output scan '") + rt_auto + "' from file '" + filename + "'");
           }
 
-          out.addSpectrum(*it2);
-          out.getSpectra().back().setRT(rt_final);
-          out.getSpectra().back().setNativeID(native_id);
-
-          if (user_ms_level)
+          spec_it->setRT(rt_final);
+          spec_it->setNativeID("spectrum=" + String(native_id));
```

Comment:
at some point we might think about whether we can preserve the native id instead of setting it to a new one - but that code is not related to your commit